### PR TITLE
Update RAJA, BLT, and GitLab CI to match RAJA

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary 
- This PR pulls in latest RAJA develop and matches the BLT version and GitLab CI files to match that.

### CODE REVIEW

File: .gitlab/jobs/lassen.yml [hunk 1]
```diff
@@ -16,10 +16,10 @@
 # Overriding shared spec: Longer allocation + extra flags
 xl_2022_08_19_gcc_8_3_1_cuda_11_2_0:
   variables:
-    SPEC: "${PROJECT_LASSEN_VARIANTS} +cuda cxxflags==\"-qthreaded -std=c++14 -O3 -qstrict -qxlcompatmacros -qlanglvl=extended0x -qalias=noansi -qhot -qpic -qsmp=omp -qsuppress=1500-029 -qsuppress=1500-036\" %xl@16.1.1.12.gcc.8.3.1 ^cuda@11.2.0+allow-unsupported-compilers ${PROJECT_LASSEN_DEPS}"
+    SPEC: "${PROJECT_LASSEN_VARIANTS} +cuda cxxflags==\"-qthreaded -std=c++14 -O3 -qstrict -qxlcompatmacros -qlanglvl=extended0x -qalias=noansi -qhot -qpic -qsmp=omp -qsuppress=1500-029 -qsuppress=1500-036\" %xl@16.1.1.12.gcc.8.3.1 ^cuda@11.2.0+allow-unsupported-compilers ${PROJECT_LASSEN_DEPS} ^blt@git.a7f0a6ecc4fdfa1724399b1454c3909b9ee02e81=develop"
```
Comment (adrienbernede) [inline .gitlab/jobs/lassen.yml:19]: Per .gitlab/custom_jobs_and_variables the variable PROJECT_LASSEN_DEPS is already set to: "^blt@git.a7f0a6ecc4fdfa1724399b1454c3909b9ee02e81=develop"

**Review by adrienbernede — CHANGES_REQUESTED** [review]
Specs in overridden jobs should use the variables set in .gitlab/custom_jobs_and_variables. Setting blt hash there will also enforce the version in the shared specs.

The reference of radiuss-shared-ci should be updated in .gitlab-ci to v2023.09.0, there are changes in this PR that correspond to that release.

File: .gitlab/jobs/lassen.yml (hunk 1 above)
Comment (rhornung67) [reply .gitlab/jobs/lassen.yml:19]: I was following what is currently done in RAJA.  I will make the change when CI passes. Please let me know if I see any other issues with bad form, etc.

**Comment by rhornung67** [comment]
I think the GitLab CI failures are due to some funkiness with GitHub communicating with LC. If I restart a job, it fails immediately, but I can't see the log. I will restart later.

**Comment by adrienbernede** [comment]
It looks like I had forgotten to update RAJAPerf to Shared CI September release, which was more intrusive than usual...

**Comment by rhornung67** [comment]
> It looks like I had forgotten to update RAJAPerf to Shared CI September release, which was more intrusive than usual...

No worries. Thanks for catching that. I am fixing more things as I figure out the new organization.

File: .gitlab/jobs/lassen.yml (hunk 1 above)
Comment (rhornung67) [reply .gitlab/jobs/lassen.yml]: @adrienbernede I think I've addressed your comments about applying the proper spec conventions in overridden vs. extra jobs.

**Review by adrienbernede — APPROVED** [review]

**Comment by rhornung67** [comment]
@MrBurmark @artv3 @rchen20 do any of you want to look over this PR before I merge it?

**Review by MrBurmark — APPROVED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -6,72 +6,82 @@
...
 # at Lawrence Livermore National Laboratory (LLNL).
-#
 # This entire pipeline is LLNL-specific
 #
-# Important note: This file is a template provided by
-# llnl/radiuss-shared-ci. Changes needed consists in setting variable values,
-# change the reference to the radiuss-shared-ci repo, opt-in and out optional
-# features. The project can then extend it with additional stages.
+# Important note: This file is a template provided by llnl/radiuss-shared-ci.
+# Remains to set variable values, change the reference to the radiuss-shared-ci
```
Comment (rchen20) [inline .gitlab-ci.yml:16]: `Remains to set` is a bit awkward, and I'm not sure how this should be rephrased.

**Review by rchen20 — APPROVED** [review]

**Pull request merged**